### PR TITLE
chore(API): Skip workspace creation when user is admin

### LINKED ISSUE #191: API: Skip workspace creation if the user is an admin user
## Description
Any user that gets created, gets a default workspace associated with them. We don't want admin accounts to get a default workspace. 

## Solution

- Add a `if` check in the [create-user.ts](https://github.com/keyshade-xyz/keyshade/blob/develop/apps/api/src/common/create-user.ts) file to skip creation of a workspace if the user is an admin user.
- Add relevant tests to `user.e2e.spec.ts` to check this case.

### PR DESCRIPTION
**Pull request by DhairyaMajmudar**
## Description
 Skip workspace creation when user is admin

Fixes #191 

## Dependencies
N/A

## Future Improvements
N/A

## Mentions
@rajdip-b 

## Screenshots of relevant screens
N/A

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [x] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [x] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

### CODE REVIEW

**Comment by DhairyaMajmudar** [comment]
I hope Rajdip bro is not gonna mad on me after so many mentions 🤣

File: apps/api/src/user/user.e2e.spec.ts [hunk 1]
```diff
@@ -9,7 +9,8 @@ import { AppModule } from '../app/app.module'
 import { AuthProvider, User } from '@prisma/client'
 import { MAIL_SERVICE } from '../mail/services/interface.service'
 import { MockMailService } from '../mail/services/mock.service'
-import { UserService } from './service/user.service'
+import { UserService } from './serice/user.service'
+import { create } from 'domain'
```
Comment (rajdip-b) [inline apps/api/src/user/user.e2e.spec.ts:13]: What is this?

**Review by rajdip-b — CHANGES_REQUESTED** [review]

File: apps/api/src/user/user.e2e.spec.ts (hunk 1 above)
Comment (DhairyaMajmudar) [reply apps/api/src/user/user.e2e.spec.ts:13]: oopsie : )

**Review by rajdip-b — APPROVED** [review]

**Comment by rajdip-b** [comment]
Seems like there are type errors. Can you fix them?

**Comment by DhairyaMajmudar** [comment]
> Seems like there are type errors. Can you fix them?

I've fixed the type errors : )

**Review by rajdip-b — APPROVED** [review]
LGTM!

**Pull request merged**